File: csharp/MapleStory.OpenAPI/Src/Dto/CharacterCashItemEquipmentPresetDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 프리셋 장착 캐시 장비 정보
    /// </summary>
    public class CharacterCashItemEquipmentPresetDTO
    {
        /// <summary>
        /// 캐시 장비 부위 명
        /// </summary>
        [JsonProperty("cash_item_equipment_part")]
        public string CashItemEquipmentPart { get; set; }

        /// <summary>
        /// 캐시 장비 슬롯 위치
        /// </summary>
        [JsonProperty("cash_item_equipment_slot")]
        public string CashItemEquipmentSlot { get; set; }

        /// <summary>
        /// 캐시 장비 명
        /// </summary>
        [JsonProperty("cash_item_name")]
        public string CashItemName { get; set; }

        /// <summary>
        /// 캐시 장비 아이콘
        /// </summary>
        [JsonProperty("cash_item_icon")]
        public string CashItemIcon { get; set; }

        /// <summary>
        /// 캐시 장비 설명
        /// </summary>
        [JsonProperty("cash_item_description")]
        public string? CashItemDescription { get; set; }

        /// <summary>
        /// 캐시 장비 옵션 목록
        /// </summary>
        [JsonProperty("cash_item_option")]
        public List<CharacterCashItemEquipmentOptionDTO> CashItemOption { get; set; }

        /// <summary>
        /// 캐시 장비 유효 기간 (KST)
        /// </summary>
        public DateTimeOffset? DateExpire
        {
            get
            {
                if (_dateExpire != null)
                {
                    return DateTimeOffset.Parse(_dateExpire).ToOffset(TimeSpan.FromHours(9));
                }

                return null;
            }
        }

        [JsonProperty("date_expire")]
        private string? _dateExpire { get; set; }

        /// <summary>
        /// 캐시 장비 옵션 유효 기간 (KST, 시간 단위 데이터로 분은 일괄 0으로 표기)
        /// </summary>
        public DateTimeOffset? DateOptionExpire
        {
            get
            {
                if (_dateOptionExpire != null && _dateOptionExpire != "expired")
                {
                    return DateTimeOffset.Parse(_dateOptionExpire).ToOffset(TimeSpan.FromHours(9));
                }

                return null;
            }
        }

        [JsonProperty("date_option_expire")]
        private string? _dateOptionExpire { get; set; }

        /// <summary>
        /// 캐시 장비 옵션 유효 기간 만료 여부
        /// </summary>
        public bool? IsOptionExpired
        {
            get
            {
                if (_dateOptionExpire == null)
                {
                    return null;
                }

                return _dateOptionExpire == "expired";
            }
        }

        /// <summary>
        /// 캐시 장비 라벨 정보
        /// </summary>
        [JsonProperty("cash_item_label")]
        public string? CashItemLabel { get; set; }

        /// <summary>
        /// 캐시 장비 컬러링프리즘 정보
        /// </summary>
        [JsonProperty("cash_item_coloring_prism")]
        public CharacterCashItemEquipmentColoringPrismDTO? CashItemColoringPrism { get; set; }

        /// <summary>
        /// 다른 프리셋에서 장비 추가 장착 없이 1번 프리셋의 장비 공유를 비활성화 했는지 여부
        /// </summary>
        [JsonProperty("base_preset_item_disable_flag")]
        public string BasePresetItemDisableFlag { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterCashItemEquipmentPresetDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.ZonedDateTime;
import java.util.List;

/**
 * Equipped cash item preset information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterCashItemEquipmentPresetDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterCashItemEquipmentPresetDTO<
        CharacterCashItemEquipmentOptionDTO,
        CharacterCashItemEquipmentColoringPrismDTO> {

    /**
     * Cash equipment part name
     */
    @JsonProperty("cash_item_equipment_part")
    private String cashItemEquipmentPart;

    /**
     * Cash equipment slot position
     */
    @JsonProperty("cash_item_equipment_slot")
    private String cashItemEquipmentSlot;

    /**
     * Cash equipment name
     */
    @JsonProperty("cash_item_name")
    private String cashItemName;

    /**
     * Cash equipment icon
     */
    @JsonProperty("cash_item_icon")
    private String cashItemIcon;

    /**
     * Cash equipment description
     */
    @JsonProperty("cash_item_description")
    private String cashItemDescription;

    /**
     * Cash equipment option
     */
    @JsonProperty("cash_item_option")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterCashItemEquipmentOptionDTO> cashItemOption;

    /**
     * Cash equipment validity period (SGT)
     */
    @JsonProperty("date_expire")
    private String dateExpire;

    /**
     * Cash equipment validity period (SGT)
     */
    public ZonedDateTime getDateExpire() {
        return dateExpire != null
                ? Utils.toZonedDateTime(this.dateExpire)
                : null;
    }

    /**
     * Cash equipment option validity period (SGT, data with minutes set to 0)
     */
    @JsonProperty("date_option_expire")
    private String dateOptionExpire;

    /**
     * Cash equipment option validity period (SGT, data with minutes set to 0)
     */
    public ZonedDateTime getDateOptionExpire() {
        return dateOptionExpire != null
                ? Utils.toZonedDateTime(this.dateOptionExpire)
                : null;
    }

    /**
     * Cash equipment label information
     */
    @JsonProperty("cash_item_label")
    private String cashItemLabel;

    /**
     * Cash equipment coloring prism information
     */
    @JsonProperty("cash_item_coloring_prism")
    private CharacterCashItemEquipmentColoringPrismDTO cashItemColoringPrism;

    /**
     * Gender compatibility for item equipment
     */
    @JsonProperty("item_gender")
    private String itemGender;

    /**
     * Whether the cash equipment is expired
     */
    public Boolean isExpired() {
        if (this.dateExpire == null) {
            return null;
        }
        return "expired".equals(this.dateExpire);
    }

    /**
     * Whether the cash equipment option is expired
     */
    public Boolean isOptionExpired() {
        if (this.dateOptionExpire == null) {
            return null;
        }
        return "expired".equals(this.dateOptionExpire);
    }
}
